app/extensions/regexp_extensions.rb

```diff
@@ -0,0 +1,10 @@
+# frozen_string_literal: true
+
+class Regexp
+  # Spaces are not Regexp meta-characters, so
+  # stop Ruby from escaping them
+  # Cf. https://stackoverflow.com/questions/73933394/regexp-escape-adds-weird-escapes-to-a-plain-space
+  def self.escape_except_spaces(str)
+    escape(str).gsub("\\ ", " ")
+  end
+end
```

Comment:
Should I use a different approach?
Back story:
In a couple of places, I wanted to check for the correct flash message with `assert_flash(<matcher>)`,
where <matcher> was something like `:runtime_no_matches.t(type: :observation)`.
See ObservationControllerTest#test_index_at_location_without_observations.

I wanted to use a regexp as the matcher. But the string `:runtime_no_matches.t(type: :observation)` didn't work because the flash included other stuff (in particular <p><\p>). 
I tried turning that string into a regexp by running it through `Regexp.escape` and using the result as the matcher. 
That didn't work because -- to my surprise -- `Regexp.escape` escapes spaces (in addition to meta characters).
So I created this simple `Regexp` extension.

Reply:
Joe - I wrote a string_extensions method you might be able use here. 
Check out `string_extensions#as_displayed` — it strips out html tags and extra spaces. You'd use it like `assert_flash(<matcher>.as_displayed)`.

But i'm not sure that's what you need. Is it your matcher that has the `<p>` tag, or the flash it's matching against?

Reply:
Yeah, I think (if I understand the situation) I agree with Nimmo.  I think it would be better to remove html markup from the strings before comparing them instead of messing with matchers.  Maybe.  I'm open to being convinced otherwise, though.

Reply:
Jason/Nimmo:
I don't get to remove the html markup before `flash_alert(<string or regexp>)` makes the comparison.
So the choices that jump out at me are:
- Use `flash_alert` with a regexp;
```ruby
assert_flash(flash_matcher)
```
- Use `flash_alert`, manually wrap the expected string with tags, 
```ruby
assert_flash(tag.p(:runtime_no_matches.t(type: :observation))
```
- Don't use `flash_alert`. Instead use:
```ruby
assert_equal(:runtime_no_matches.t(type: :observation),
             get_last_flash[1..-1]. # strip the error level code
             strip_html) # strip the p tags
```
What do you like best?

Reply:
P.S.: I forgot to mention one of the things that made it hard to create the matcher:
In this case, the flash message contains regexp meta characters. 
So if using a regexp as a matcher, I have to quote those characters, but not spaces.